raft/raft.go

```diff
@@ -242,6 +246,7 @@ func (r *raft) bcastHeartbeat() {
 		}
 		r.sendHeartbeat(i)
 		r.prs[i].resume()
+		r.prs[i].ins.reset()
```

Comment:
@bdarnell 
> Maybe it should reset on receiving an MsgHeartbeatResp instead of sending an MsgHeartbeat.

Maybe. It is the same in etcd's transport since we will report failure if the heartbeat fails.

> I wonder if it would be better to remove one request from inflights instead of all of them. This would guarantee that you have a minimum of one MsgApp per heartbeat interval in replicate mode, just like in probe mode. Once you have an MsgAppResp its index will clear out the rest of the inflights.

I can try this idea out. I just do not want to go to implement the full TCP, which increases the window for each RTT (in our case, each heartbeat) and decreases when there is a drop.